stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt

```diff
@@ -41,14 +49,17 @@ internal class DefaultChatEventHandler(private val client: ChatClient, private v
                 memberLimit = 0,
             )
         ).await()
-            .map { channels -> channels.any { it.cid == cid } }
-            .let { filteringResult -> filteringResult.isSuccess && filteringResult.data() }
     }
 
     override fun handleNotificationAddedToChannelEvent(
         event: NotificationAddedToChannelEvent,
         filter: FilterObject,
-    ): EventHandlingResult = handleMemberUpdate(event, event.cid, filter)
+    ): EventHandlingResult = handleMemberUpdate(event.channel, event.cid, filter)
+
+    override fun handleMemberAddedEvent(
+        event: MemberAddedEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = handleMemberUpdate(event.cid, filter)
```

Comment:
I think we shouldn't do additional requests here. Actually, we should get rid of `query channels` requests from this class and replace them with default behavior which will add or remove channels based on membership

Reply:
The problem is that we don’t know if an Add really means Add or Remove in the query, it depends on the filter. Example: 
- I’m filtering channel that I’m **not** a member.
- I receive an event NotificationAddedToChannel
- That channel will fail in my filter now, because now I’m a member.
- As a result that channel has to be removed from the list, although the event is NotificationAddedToChannel.

So, we have: 
 Added to channel -> Filter Passes -> Added to the list
 Added from channel -> Filter Fails -> Removed from the list
 Removed from the channel -> Filter Passes -> Added to the list
 Removed from the channel -> Filter Fails -> Removed to the list
 
 If we don’t use the filter, there’s no way to know which is the correct behaviour. This class would have the wrong logic without it =|.

Reply:
Unfortunately, we don't have a good way to solve it right now and that's why we decided to leave this decision to integrators. They are providing the filter, so they should know what to do in such a case. It's not the best possible solution, but an approach with querying channels is not safe.
For now, we should have a solid default implementation with good docs and we'll revisit this case once we have a better solution

Reply:
as we discussed, I added a default implementation with no request to channels. And there's a factory to use different implementations of the event handler and an ChannelEventHandler to to list channels that we user is not a member. (ChannelEventHandler)